src/worker/abstract-worker.ts

```diff
@@ -43,20 +62,42 @@ export abstract class AbstractWorker<
     }
   }
 
+  /**
+   * Returns the main worker.
+   */
   protected abstract getMainWorker (): MainWorker
 
+  /**
+   * Send a message to the main worker.
+   *
+   * @param message The response message.
+   */
   protected abstract sendToMainWorker (message: MessageValue<Response>): void
 
+  /**
+   * Check to see if the worker should be terminated, because its living too long.
+   */
   protected checkAlive (): void {
     if (Date.now() - this.lastTask > this.maxInactiveTime) {
       this.sendToMainWorker({ kill: 1 })
     }
   }
 
+  /**
+   * Handle an error and convert it to a string so it can be sent back to the main worker.
+   *
+   * @param e The error raised by the worker.
+   */
   protected handleError (e: Error | string): string {
     return (e as unknown) as string
```

Comment:
We only do that for the cluster worker right ?

Reply:
We override it only in the cluster pool.
The thread pool uses the default impl as can be seen here.

The JSDoc lies a bit :sweat_smile:  
Because here we could receive an `Error` and just lie to TS and tell TS that the `Error` is a `string`
You can do that with `(something as unknown) as OtherType`
That is a workaround to tell TS to shut the **** up :rofl: